tests/std/tests/P2441R2_views_join_with/test.cpp

```diff
@@ -558,6 +560,76 @@ void test_valueless_iterator() {
     }
 }
 
+constexpr bool test_lwg3698() {
+    // LWG-3698 "regex_iterator and join_view don't work together very well"
+    struct stashing_iterator {
+        using difference_type = int;
+        using value_type      = span<const int>;
+
+        int x = 1;
+
+        constexpr stashing_iterator& operator++() {
+            ++x;
+            return *this;
+        }
+        constexpr void operator++(int) {
+            ++x;
+        }
+        constexpr value_type operator*() const {
+            return {&x, &x + 1};
+        }
+        constexpr bool operator==(default_sentinel_t) const {
+            return x > 3;
+        }
+    };
+
+    auto r   = ranges::subrange{stashing_iterator{}, default_sentinel} | views::join_with(views::empty<int>);
+    auto r2  = r;
+    auto it  = r.begin();
+    auto it2 = r2.begin();
+
+    auto itcopy = it;
+    it          = ++it2;
+    assert(*itcopy == 1);
+
+    int expected_ints[]          = {1, 2, 3, 5, 7};

```

Comment:
These aren't modified, and they're used with `span<const int>`. I strongly believe that these should be `const` or `constexpr`. (Test code doesn't need to adhere to the highest levels of `const` cleanliness, but it's nice to know that our expected values won't change.)